zerver/lib/user_groups.py

```diff
@@ -293,13 +293,13 @@ def get_direct_user_groups(user_profile: UserProfile) -> List[UserGroup]:
 def get_user_group_direct_member_ids(
     user_group: UserGroup,
 ) -> ValuesQuerySet[UserGroupMembership, int]:
-    return UserGroupMembership.objects.filter(user_group=user_group).values_list(
-        "user_profile_id", flat=True
-    )
+    return UserGroupMembership.objects.filter(
+        user_group=user_group, user_profile__is_active=True
+    ).values_list("user_profile_id", flat=True)
 
 
 def get_user_group_direct_members(user_group: UserGroup) -> QuerySet[UserProfile]:
-    return user_group.direct_members.all()
+    return user_group.direct_members.filter(is_active=True)
 
 
 def get_direct_memberships_of_users(user_group: UserGroup, members: List[UserProfile]) -> List[int]:
```

Comment:
Have not added `user_profile__is_active=True` condition here since we expect the `members` list to only contain active users.

Reply:
That seems OK. If you call this with a deactivated user, it seems like you . Could have a comment, like:

> Returns the subset of the provided `members` list who are direct subscribers of the group. If a deactivated user is passed, it will be returned if the user has was a member of the group when deactivated.